src/main/java/ch/njol/skript/expressions/ExprSeaPickles.java

```diff
@@ -0,0 +1,151 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.block.data.type.SeaPickle;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Sea Pickles")
+@Description("An expression to obtain or modify data relating to the pickles of a sea pickle block.")
+@Examples({
+	"on block break:",
+	"\ttype of block is sea pickle",
+	"\tsend \"Wow! This stack of sea pickles contained %event-block's sea pickle count% pickles!\"",
+	"\tsend \"It could've contained a maximum of %event-block's maximum sea pickle count% pickles!\"",
+	"\tsend \"It had to have contained at least %event-block's minimum sea pickle count% pickles!\"",
+	"\tcancel event",
+	"\tset event-block's sea pickle count to event-block's maximum sea pickle count",
+	"\tsend \"This bad boy is going to hold so many pickles now!!\""
+})
+@Since("INSERT VERSION")
+public class ExprSeaPickles extends SimplePropertyExpression<Block, Integer> {
+
+	static {
+		register(ExprSeaPickles.class, Integer.class, "[:(min|max)[imum]] [sea] pickle(s| (count|amount))", "blocks");
+	}
+
+	private boolean minimum, maximum;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		minimum = parseResult.hasTag("min");
+		maximum = parseResult.hasTag("max");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Block block) {
+		BlockData blockData = block.getBlockData();
+		if (!(blockData instanceof SeaPickle))
+			return null;
+
+		SeaPickle pickleData = (SeaPickle) blockData;
+
+		if (maximum)
+			return pickleData.getMaximumPickles();
+		if (minimum)
+			return pickleData.getMinimumPickles();
+		return pickleData.getPickles();
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
```

Comment:
```suggestion
	@Override
	@Nullable
	public Class<?> acceptChange(ChangeMode mode) {
```

Reply:
You have potential for returning just null.

Reply:
I had future proofed this since the parent annotation is wrong (right now it's annotated that the items can be nullable which is NOT true, only the array itself can be null). However, I can remove this if you want, and it can be changed in the future when annotations are fixed.

Reply:
You can literally return null in acceptChange

Reply:
I was futureproofing this, but I will remove it for now while we determine how to handle the annotation changes that will be needed.